### PR TITLE
fix: add missing importlib.resouces.as_file for Python 3.9

### PR DESCRIPTION
**Pull request by henryiii**
As seen https://docs.python.org/3/library/importlib.html#importlib.resources.as_file, for Python 3.9.

### CODE REVIEW

File: stdlib/importlib/resources.pyi [hunk 1]
```diff
@@ -19,5 +19,8 @@ if sys.version_info >= (3, 7):
     def contents(package: Package) -> Iterator[str]: ...
 
 if sys.version_info >= (3, 9):
+    from contextlib import AbstractContextManager
     from importlib.abc import Traversable
+
```
Comment (henryiii) [inline stdlib/importlib/resources.pyi:24]: Weird, wonder why the extra line is fine above for black, but not here? Will fix.

**Review by hauntsaninja — COMMENTED** [review]
pathlib.Path should be a Traversable, so if everything is working as intended, you shouldn't need to specify it in the argument union.
Also nit: in typeshed it looks like we prefer typing.ContextManager over contextlib.AbstractContextManager

**Comment by henryiii** [comment]
> in typeshed it looks like we prefer typing.ContextManager over contextlib.AbstractContextManager

This is because `contextlib.AbstractContextManager` didn't work like this until Python 3.9. But this is already protected for 3.9+ only, so it should be fine. (and [ContextManager is deprecated](https://docs.python.org/3/library/typing.html#typing.ContextManager))

**Comment by hauntsaninja** [comment]
Thanks!

**Pull request merged**